internal/command/jsonplan/plan.go

```diff
@@ -758,6 +758,49 @@ func (p *plan) marshalRelevantAttrs(plan *plans.Plan) error {
 
 		p.RelevantAttributes = append(p.RelevantAttributes, ResourceAttr{addr, path})
 	}
+
+	// we want our outputs to be deterministic, so we'll sort the attributes
+	// here. The order of the attributes is not important, as long as it is
+	// stable.
+
+	sort.SliceStable(plan.RelevantAttributes, func(i, j int) bool {
+		if !plan.RelevantAttributes[i].Resource.Equal(plan.RelevantAttributes[j].Resource) {
+			return plan.RelevantAttributes[i].Resource.Less(plan.RelevantAttributes[j].Resource)
+		}
+
+		str := func(step cty.PathStep) string {
```

Comment:
This is only called once when writing the plan, so you could probably get away with just comparing `fmt.Sprintf("%#v", attr)` of the whole `ResourceAttr`. You've already written the sort though, so no need to remove it unless you really don't like it ;)

Reply:
Yes, that is much simpler!